deployment/configenv/xml_jlibpt/SWProcess.hpp

```diff
@@ -0,0 +1,76 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+#ifndef _SWPROCESS_HPP_
+#define _SWPROCESS_HPP_
+
+#include "EnvHelper.hpp"
+#include "ComponentBase.hpp"
+#include "SWComponentBase.hpp"
+
+namespace ech
+{
+
+class SWProcess : public SWComponentBase
+{
+public:
+   SWProcess(const char* name, EnvHelper * envHelper);
+   virtual ~SWProcess() {}
+
+   virtual void create(IPropertyTree *params);
+   virtual unsigned add(IPropertyTree *params);
+   virtual void modify(IPropertyTree *params);
+   //virtual void remove(IPropertyTree *params);
+   virtual IPropertyTree * addComponent(IPropertyTree *params);
+   virtual IPropertyTree * cloneComponent(IPropertyTree *params);
+
+   virtual void computerAdded(IPropertyTree *computerNode, const char* instanceXMLTagName=NULL);
+   virtual void computerUpdated(IPropertyTree *computerNode, const char *oldName, const char *oldIp, const char* instanceXMLTagName=NULL);
```

Comment:
It's better to start using null_ptr instead of NULL

Reply:
Agree - new code should be using null_ptr